lib/vfscore/volumes.c

```diff
@@ -0,0 +1,59 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <string.h>
+#include <sys/mount.h>
+#include <uk/init.h>
+#include <uk/libparam.h>
+
+#define UK_MAX_VFSCORE_VOLUMES_COUNT			64
+#define UK_VFSCORE_VOLUMES_ARGS_SEP			":"
+
+static char *vfscore_volumes[UK_MAX_VFSCORE_VOLUMES_COUNT];
+
+UK_LIBPARAM_PARAM_ARR_ALIAS(automount, &vfscore_volumes, charp,
+			    UK_MAX_VFSCORE_VOLUMES_COUNT, "VFSCore Volumes");
+
+/**
+ * Expected command-line argument format:
+ *	vfs.automount=[
+ *		"<source_device>:<mount_target>:<fs_driver>:<options>"
+ *		"<source_device>:<mount_target>:<fs_driver>:<options>"
+ *		...
+ *	]
+ */
+static void vfscore_volume_fetch_args(char *v, char **sdev,
+				      char **path, char **drv,
+				      char **opts)
+{
+	*sdev = strtok(v, UK_VFSCORE_VOLUMES_ARGS_SEP);
+	*path = strtok(NULL, UK_VFSCORE_VOLUMES_ARGS_SEP);
+	*drv = strtok(NULL, UK_VFSCORE_VOLUMES_ARGS_SEP);
+	*opts = strtok(NULL, UK_VFSCORE_VOLUMES_ARGS_SEP);
+}
+
+static int vfscore_volumes_mount(void)
+{
+	char *sdev, *path, *drv, *opts;
+	int rc, i;
+
+	for (i = 0; i < UK_MAX_VFSCORE_VOLUMES_COUNT &&
+	     vfscore_volumes[i]; i++) {
+		vfscore_volume_fetch_args(vfscore_volumes[i], &sdev, &path,
+					  &drv, &opts);
+		uk_pr_info("Mount %s to %s...\n", sdev, path);
+
+		/* For now, mount() flags are set to 0 */
+		rc = mount(sdev, path, drv, 0, opts);
+		if (unlikely(rc))
+			uk_pr_err("Failed to mount %s: error %d\n", sdev, rc);
+	}
+
+	return 0;
+}
+
+/* Should be preferrably called after vfscore's rootfs is mounted (prio 4) */
```

Comment:
Could you elaborate a little bit about priorities. Why it should be 4 here?

Reply:
What I meant was that `rootfs.c` has a `uk_rootfs_initcall_prio(vfscore_rootfs, 4);`. Therefore, in order to ensure this can only be called after an explicit rootfs, give this a priority of `4 + 1 = 5`. I was referring to the `rootfs`. I will make the comment more clear :100:.